src/marketplace/localserver.ts

```diff
@@ -0,0 +1,111 @@
+import * as bodyParser from "body-parser";
+import * as express from "express";
+import * as http from "http";
+import * as request from "request-promise";
+import * as vscode from "vscode";
+import { Utility } from "../common/utility";
+
+export class LocalServer {
+    private app: express.Express;
+    private server: http.Server;
+    private serverPort = 0;
+    private router: express.Router;
+    private context: vscode.ExtensionContext;
+    private modules;
+
+    constructor(context: vscode.ExtensionContext, modules?: string[]) {
+        this.initRouter();
+        this.initApp();
+        this.server = http.createServer(this.app);
+        this.context = context;
+        this.modules = modules;
+    }
+
+    public startServer(): void {
+        const port = this.server.listen(0).address().port;
+        this.serverPort = port;
+        // tslint:disable-next-line:no-console
+        console.log("serverPort:" + this.serverPort);
+    }
+
+    public stopServer(): void {
+        this.server.close(null);
+    }
+
+    public getServerUri(): string {
+        return `http://localhost:${this.serverPort}`;
+    }
+
+    private initRouter() {
+        this.router = express.Router();
+        this.router.get("/api/v1/modules", async (req, res, next) => await this.getModules(req, res, next));
+        this.router.get("/api/v1/modules/:module/status", async (req, res, next) => await this.validateModuleName(req, res, next));
```

Comment:
this api seems should not belong to the server

Reply:
valid module name need to read deployment.template.json, so we need to put it in the server

Reply:
it could be put in the message handler. The point here is the front could just trigger a message with the necessary meta data and ask vscode to do the import including validation.